src/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,55 @@
+import { getUserNameFromEmail } from "@/utils/getUserNameFromEmail";
+import React, { createContext, useContext, useState } from "react";
+
+type UserInfo = {
+  email: string;
+  name: string;
+  authToken: string;
+};
+
+type AuthContextType = {
+  isLoggedIn: boolean;
+  userInfo: UserInfo | null;
+  login: (userInfo: UserInfo) => void;
+  logout: () => void;
+};
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+export const AuthProvider = ({ children }: { children: React.ReactNode }) => {
+  const [isLoggedIn, setIsLoggedIn] = useState<boolean>(() => {
+    return localStorage.getItem("isLoggedIn") === "true";
+  });
+  const [userInfo, setUserInfo] = useState<UserInfo | null>(() => {
+    const stored = localStorage.getItem("userInfo");
+    return stored ? JSON.parse(stored) : null;
+  });
```

Comment:
```suggestion
  const [loggedIn, setLoggedIn] = useStorageState('isLoggedIn');
  const [userInfo, setUserInfo] = useStorageState('userInfo');
```

이렇게 쓰일 수 있도록 customHook을 하나 정의해주면 좋겠어요!

그리고 스토리지가 분리되어 쓰이고 있는데, 하나로 쓰여도 무방하지 않을까요?
아니면 파생상태로 관리한다거나...